crates/bevy_scene/src/serde.rs

```diff
@@ -174,22 +192,35 @@ impl<'a, 'de> Visitor<'de> for SceneVisitor<'a> {
             }
         }
 
+        let resources = resources.ok_or_else(|| Error::missing_field(SCENE_RESOURCES))?;
```

Comment:
If you're wanting to make this backwards-compatible, you should be able to remove this line. 

> Note: this should only work for non-binary data

Reply:
As specified in the edited PR description, I think such a point would require futher discussion if it's something we actually want, since as you said it might be quite hard to do for binary serialization.